### PR TITLE
http2tcp: init at 0.5

### PR DESCRIPTION
**Pull request by clkamp**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/wsgitools/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+buildPythonPackage rec {
+  pname = "wsgitools";
+  rev = "44bdfe2a595ce06dfda3d00ee0cf65e565c3cbea";
+  version = "0.3postgit20190310";
+
+  src = fetchgit {
+    inherit rev;
+    url = "git://git.subdivi.de/~helmut/wsgitools.git";
+    sha256 = "17i939zlk5k1hai51qyhm234fi7f0xc83qgh14r396khg9zsjjnw";
+  };
```
Comment (jonringer) [inline pkgs/development/python-modules/wsgitools/default.nix]: can they do a new release on pypi? https://pypi.org/project/wsgitools/ seems 5 years out of date anyway

Comment (clkamp) [reply pkgs/development/python-modules/wsgitools/default.nix]: For http2tcp there is already a new release, for wsgitools I have asked for one.

Comment (clkamp) [reply pkgs/development/python-modules/wsgitools/default.nix]: For http2tcp there is already a new release, for wsgitools I have asked for one.

Comment (clkamp) [reply pkgs/development/python-modules/wsgitools/default.nix]: Now wsgitools comes from pypi at 0.3.1

File: pkgs/tools/networking/http2tcp/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{ lib
+, python3Packages
+, fetchurl
+}:
+
+python3Packages.buildPythonApplication rec {
+
```
Comment (jonringer) [inline pkgs/tools/networking/http2tcp/default.nix]:
```suggestion
```

File: pkgs/tools/networking/http2tcp/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  pname = "http2tcp";
+  version = "0.5";
```
Comment (jonringer) [inline pkgs/tools/networking/http2tcp/default.nix]: i would move format up here
```suggestion
  version = "0.5";
  format = "other";
```

File: pkgs/tools/networking/http2tcp/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  src = fetchurl {
+    url = "https://www.linta.de/~aehlig/http2tcp/${pname}-${version}.tar.gz";
+    sha256 = "34fb83c091689dee398ca80db76487e0c39abb17cef390d845ffd888009a5caa";
+  };
+
+  propagatedBuildInputs = with python3Packages; [ wsgitools ];
+
+  format = "other";
```
Comment (jonringer) [inline pkgs/tools/networking/http2tcp/default.nix]:
```suggestion
```

File: pkgs/tools/networking/http2tcp/default.nix [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/{bin,share/${pname}}
+    cp http2tcp* $out/bin
+    cp Protocol $out/share/${pname}/
+  '';
```
Comment (jonringer) [inline pkgs/tools/networking/http2tcp/default.nix]: this doesn't look to be a normal python installation, you probably want to wrap the program with a python interpreter using python.withPackage https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#pythonwithpackages-function

Comment (clkamp) [reply pkgs/tools/networking/http2tcp/default.nix]: I have changed it that way

**Review by jonringer — APPROVED** [review]
diff LGTM
shows usage

```
[7 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/81895
4 package built:
http2tcp python27Packages.wsgitools python37Packages.wsgitools python38Packages.wsgitools
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build http2tcp python27Packages.wsgitools python37Packages.wsgitools python38Packages.wsgitools

**Pull request merged**